doc.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// yab is a benchmarking tool for TChannel and HTTP applications.
+// It's primarily intended for Thrift applications but supports other encodings like JSON and binary (raw).
+//
+// It can be used in a curl-like fashion when benchmarking features are disabled.
+//
+// For usage information, check out the man page:
+// http://yarpc.github.io/yab/man.html
+package main
+
+const _reqOptsDesc = `Configures the request data and the encoding.
+
+To make Thrift requests, specify a Thrift file and pass the Thrift
+service and procedure to the method argument (-m) as Service::Method.
+
+The TChannel health endpoint can be hit without specifying a Thrift file
+by passing --health.
+
+Thrift requests can be specified as JSON or YAML. For example, for a method
+defined as:
+  void addUser(1: string name, 2: i32 age)
+
+You can pass the request as JSON: {"name": "Prashant", age: 10}
+or as YAML:
+  name: Prashant
+	age: 10
+
+The request can be specified on the command line using -r or --request:
+  $ yab -p localhost:9787 kv KeyValue::Get -3 '{"key": "hello"}'
+
+Or it can be loaded from a file using -f or --file:
+  $ yab -p localhost:9787 kv KeyValue::Get -f req.yaml
```

Comment:
Added example with `-m` and explained how positional args can be used too. Trying to reduce the amount of noise.